pinot-broker/src/main/java/org/apache/pinot/broker/queryquota/HelixExternalViewBasedQueryQuotaManager.java

```diff
@@ -458,6 +566,27 @@ public void processQueryRateLimitingExternalViewChange(ExternalView currentBroke
             numRebuilt, _rateLimiterMap.size());
   }
 
+  /**
+   * Process query quota state change when cluster config gets changed
+   */
+  public void processQueryRateLimitingClusterConfigChange() {
+    double oldDatabaseQpsQuota = _defaultQpsQuotaForDatabase;
+    getDefaultQueryQuotaForDatabase();
+    if (oldDatabaseQpsQuota == _defaultQpsQuotaForDatabase) {
+      return;
+    }
+    createOrUpdateDatabaseRateLimiter(new ArrayList<>(_databaseRateLimiterMap.keySet()));
+  }
+
+  private void getDefaultQueryQuotaForDatabase() {
```

Comment:
IMO this part is not as explicit / unambiguous to read as it could be - WDYT about making `getDefaultQueryQuotaForDatabase` return the default value and setting `_defaultQpsQuotaForDatabase` at the method's call-sites rather than within the method itself?

Reply:
Makes sense. Will make the changes👍